readthedocs/notifications/querysets.py

```diff
@@ -66,11 +66,15 @@ def cancel(self, message_id, attached_to):
             modified=timezone.now(),
         )
 
-    def for_user(self, user):
+    def for_user(self, user, resource=None):
         """
-        Retrieve all notifications related to a particular user.
+        Retrieve notifications related to resource for a particular user.
 
-        Given a user, returns all the notifications that:
+        Given a user, returns all the notifications for the specified ``resource``
+        considering permissions (e.g. does not return any notification if the ``user``
+        doesn't have admin permissions on the ``resource``).
+
+        If ``resource="all"``, it returns the following notifications:
```

Comment:
What happens if `resource` is None? If it just returns `none()`, it should probably be a required field, or default to `all`.

Reply:
Yeah, it returns an empty queryset. I don't remember why I made this decision 🤔  (maybe because I was already sick 🙃 ), but I think it makes more sense to make it a required argument, yeah.